File: 高德地图/officialDemoNavi/User/UserViewController.m
//
//  UserViewController.m
//  SF_GaoDeMAP
//
//  Created by李晓东 on 2017/12/14.
//  Copyright © 2017年 ShouNew.com. All rights reserved.
//

#import "UserViewController.h"
#import "UserCell.h"
#import "MessageViewController.h"
static NSString *u_cell = @"UserCell";
@interface UserViewController ()<UITableViewDelegate,UITableViewDataSource>
@property(nonatomic,strong)UITableView *tableView;//表
@property(nonatomic,strong)NSArray *dataArray;//数据
@end

@implementation UserViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"我的";
    //加载数据表
    self.tableView = [[UITableView alloc]initWithFrame:self.view.bounds];
    self.tableView.delegate = self;
    self.tableView.dataSource = self;
    [self.tableView registerNib:[UINib nibWithNibName:u_cell bundle:nil] forCellReuseIdentifier:u_cell];
    [self.view addSubview:self.tableView];
    [self loadHeaderView];
    
    self.dataArray = @[@"信息中心",@"行走助手",@"反馈",@"关于",@"设置",@"退出"];
}
//界面显示的时候
- (void)viewWillAppear:(BOOL)animated{
    
    [super viewWillAppear:animated];
    self.navigationController.navigationBarHidden = NO;
}
//界面消失的时候
- (void)viewWillDisappear:(BOOL)animated{
    
    [super viewWillDisappear:animated];
    self.navigationController.navigationBarHidden = YES;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    
}

//添加表头
- (void)loadHeaderView{
    
    UIView *headerView = [[UIView alloc]initWithFrame:CGRectMake(0, 0, CGRectGetWidth(self.view.frame), 200)];
    self.tableView.tableHeaderView = headerView;
    //头像
    UIImageView *icon = [[UIImageView alloc]initWithFrame:CGRectMake((CGRectGetWidth(self.view.frame) - 100) / 2, 50, 100, 100)];
    icon.image = [UIImage imageNamed:@"user"];
    icon.layer.cornerRadius= 50;
    icon.layer.cornerRadius = 5;
    icon.layer.shadowOpacity = 0.5;// 阴影透明度
    icon.layer.shadowColor = [UIColor lightGrayColor].CGColor;// 阴影的颜色
    icon.layer.shadowRadius = 3;// 阴影扩散的范围控制
    [headerView addSubview:icon];
    
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    
    return self.dataArray.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    
    UserCell *cell = [tableView dequeueReusableCellWithIdentifier:u_cell forIndexPath:indexPath];
    cell.nameLabel.text = self.dataArray[indexPath.row];
    cell.icon.image = [UIImage imageNamed:@"历史"];
    return cell;
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath{
    
    return 44.0f;
}
@end

File: My_News_Swift/My_News_Swift/Classeas/Module/User/Controller/UserViewController.swift
//
//  UserViewController.swift
//  My_News_Swift
//
//  Created by LG on 2017/9/27.
//  Copyright © 2017年 LG. All rights reserved.
//

import UIKit
import SnapKit
import SVProgressHUD
class UserViewController: UIViewController ,ExpanHeaderViewDelegate {
    
    var tableView = UITableView()
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        setCreateTableView()
        
        NotificationCenter.default.addObserver(self, selector: #selector(isLogin), name: NSNotification.Name(rawValue: LoginNsnotifaction), object: nil)
    }
    
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        self.navigationController?.setNavigationBarHidden(true, animated: false)
        isLogin()
        
        for view in (Window?.subviews)! {
            if view.isKind(of: UIButton.classForCoder()) {
                view.isHidden = true
            }
        }
        
    }
    
   @objc func isLogin() {
        
        UserServer.shared.loginActive { (userStateModel) in
            print(userStateModel)
            guard userStateModel.active == true else {
                self.expanHeadViews.roloadDate(state: false , model: userStateModel)
                return
            }
            self.expanHeadViews.roloadDate(state: true , model: userStateModel)
        }
    }
    
    func didTouchUpinSides(type: NSInteger) {

        guard type == 3 else {
             // 普通登录
            let viewC = LoginViewController()
            self.present(viewC, animated: true, completion: nil)
            return
        }
         // qq 登录
        LGTencentLogin.login()
        

    }
    
     func setCreateTableView () {
        
        let tableview = UITableView.init(frame: CGRect.init(), style: .grouped)
        tableview.delegate = self
        tableview.dataSource = self
        tableview.showsVerticalScrollIndicator = true
        tableview.showsHorizontalScrollIndicator = true
        tableview.register(MyTableViewCell.classForCoder(), forCellReuseIdentifier: MyTableViewCell.identfier!)
        tableview.separatorStyle = .none
        
        self.view.addSubview(tableview)
        self.tableView = tableview
        self.tableView.snp_makeConstraints { (make) in
            make.center.equalTo(self.view)
            make.size.equalTo(self.view)
        }
         self.tableView.addSubview(self.expanHeadViews)
    }
    
    
    lazy var expanHeadViews : ExpandHeaderView = {
        
        var exV = ExpandHeaderView()
        exV.delegate = self
        exV.expandHeaderWithScorollView(scrollview:self.tableView)
        return exV
        
        
    }()
    
    
    lazy var items : NSArray = {
        var items = NSArray()
        let cachePath = "/Users/lg/Desktop/Me-News/Swift/My_News_Swift/My_News_Swift/My.plist"
        let bundle = Bundle.main
        let myItem:String = "My"
        let plistPath = bundle.path(forResource: myItem, ofType: "plist")
        items = NSArray.init(contentsOfFile: plistPath!)!
        var arrayM = NSMutableArray()
        for array in items  {
            arrayM.add(array)
        }
        return arrayM
    }()
}

extension UserViewController : UITableViewDelegate, UITableViewDataSource , SettingViewControllerDelegate , MyHeadViewDelegate{
    
    func clickCollectTouchUpInside() {
        print("收藏")
        let collecV = CollectViewController()
        self.navigationController?.pushViewController(collecV, animated: true)
    }
    
    func didExitLogin() {
         self.expanHeadViews.roloadDate(state: false,model: UserCenter())
         LGTencentLogin.exLogin()
    }
    
    
    
    func numberOfSections(in tableView: UITableView) -> Int {
        return self.items.count
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        let item = self.items[section] as! NSArray
        print(item.count)
        return item.count
    }
    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
        if(section == 0){
          let view = MyHeadView()
          view.delegate = self
          return view
        }
        return UIView()
    }
    
    func tableView(_ tableView: UITableView, viewForFooterInSection section: Int) -> UIView? {
        return UIView()
        
    }
    
    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
        if section == 0{
            return 75
        }
        return 15
    }
    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
        return 0.01
    }
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 44
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        var cell = MyTableViewCell()
        cell = tableView.dequeueReusableCell(withIdentifier:MyTableViewCell.identfier!) as! MyTableViewCell
        let arrayM = self.items[indexPath.section] as! NSArray
        let dict = arrayM[indexPath.row] as! NSDictionary
        cell.dict = dict
        return cell
    }
    
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        if indexPath.section == 2 {
            if indexPath.row == 2 {

                let settingView = SettingViewController()
                settingView.delegate = self
                self.navigationController?.pushViewController(settingView, animated: true)
                
            }
        }
    }
    
}
